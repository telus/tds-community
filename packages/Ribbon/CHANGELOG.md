# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.2.0](https://github.com/telus/tds-community/compare/@tds/community-ribbon@1.1.0...@tds/community-ribbon@1.2.0) (2021-09-21)


### Features

* **community-ribbon:** react 17 and styled components 5 ([3143b5a](https://github.com/telus/tds-community/commit/3143b5a510bbffc3f445913173183bd706a34a29))





# [1.1.0](https://github.com/telus/tds-community/compare/@tds/community-ribbon@1.0.4...@tds/community-ribbon@1.1.0) (2021-07-14)


### Features

* **community-ribbon:** ribbon can be place on all background colors ([c39d348](https://github.com/telus/tds-community/commit/c39d348e69a8c17d91e596ff2d0e7f98dbfb6682))





## [1.0.4](https://github.com/telus/tds-community/compare/@tds/community-ribbon@1.0.3...@tds/community-ribbon@1.0.4) (2020-09-09)


### Bug Fixes

* **community-ribbon:** update util-helpers version ([ec0fbea](https://github.com/telus/tds-community/commit/ec0fbeade8d4877789ac0809ad05cbf473bb0942))





## [1.0.3](https://github.com/telus/tds-community/compare/@tds/community-ribbon@1.0.2...@tds/community-ribbon@1.0.3) (2020-07-31)

**Note:** Version bump only for package @tds/community-ribbon





## [1.0.2](https://github.com/telus/tds-community/compare/@tds/community-ribbon@1.0.1...@tds/community-ribbon@1.0.2) (2020-03-19)

**Note:** Version bump only for package @tds/community-ribbon





## [1.0.1](https://github.com/telus/tds-community/compare/@tds/community-ribbon@1.0.0...@tds/community-ribbon@1.0.1) (2019-09-17)

**Note:** Version bump only for package @tds/community-ribbon





# 1.0.0 (2019-06-13)

**Note:** Version bump only for package @tds/community-ribbon
