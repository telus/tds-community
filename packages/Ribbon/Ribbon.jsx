import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '@tds/shared-safe-rest'

import RibbonWrapper from './styles'

/** Ribbon component for focusing attention onto a sale or special feature.
 * @version ./package.json
 */
const Ribbon = ({ ribbonCopy, ...rest }) => {
  return (
    <React.Fragment>
      <RibbonWrapper {...safeRest(rest)}>{ribbonCopy}</RibbonWrapper>
    </React.Fragment>
  )
}

Ribbon.propTypes = {
  ribbonCopy: PropTypes.string,
}

Ribbon.defaultProps = {
  ribbonCopy: '',
}

export default Ribbon
