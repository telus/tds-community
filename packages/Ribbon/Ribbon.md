`Ribbon` displays promotional content.

```jsx
<Ribbon>Telus Ribbon</Ribbon>
```
