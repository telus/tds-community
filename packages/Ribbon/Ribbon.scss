.ribbonStyle {
  background: #4B286D;
  color: #FFFFFF;
  padding: 8px 30px 8px 15px;
  margin: 16px 0;
  position: relative;
  font-size: 1rem;
  display: inline-block;
  max-width: 100%;
  min-width: 126px;
  min-height: 40px;

  &:after {
    content: '';
    position: absolute;
    right: 0;
    bottom: 0;
    width: 0;
    height: 0;
    border-right: 13px solid white;
    border-top: 20px solid transparent;
    border-bottom: 20px solid transparent;
  };
};
