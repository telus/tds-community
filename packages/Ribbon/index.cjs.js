require('./dist/index.css')
const Ribbon = require('./dist/index.cjs')

module.exports = Ribbon
