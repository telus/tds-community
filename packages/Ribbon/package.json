{
  "name": "@tds/community-ribbon",
  "version": "1.0.0",
  "description": "",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telus/tds-community.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telus/tds-community/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "prop-types": "^15.6.2"
  },
  "devDependencies": {
    "@tds/shared-safe-rest": "^1.0.0"
  }
}
