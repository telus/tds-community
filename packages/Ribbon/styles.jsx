import styled from 'styled-components'

const RibbonWrapper = styled.div`
  background: #4b286d;
  color: #ffffff;
  padding: 8px 30px 8px 15px;
  margin: 16px 0;
  position: relative;
  font-size: 1rem;
  display: inline-block;
  max-width: 100%;
  min-width: 126px;
  height: 40px;
  &:after {
    content: '';
    position: absolute;
    right: 0;
    bottom: 0;
    width: 0;
    height: 0;
    border-right: 13px solid white;
    border-top: 20px solid transparent;
    border-bottom: 20px solid transparent;
  }
`

export default RibbonWrapper
