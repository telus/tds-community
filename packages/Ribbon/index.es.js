import Ribbon from './dist/index.es'

export default Ribbon
