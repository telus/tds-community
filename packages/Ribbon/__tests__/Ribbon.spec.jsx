import React from 'react'
import { shallow } from 'enzyme'

import Ribbon from '../Ribbon'

describe('Ribbon', () => {
  const doShallow = (
    props = {
      ribbonCopy: 'Prop Text',
    }
  ) => shallow(<Ribbon {...props} />)

  it('matches snapshot with ribbonCopy prop', () => {
    const ribbon = doShallow()
    expect(ribbon).toMatchSnapshot()
  })

  it('matches snapshot with empty ribbonCopy prop', () => {
    const ribbon = doShallow()
    ribbon.setProps({ ribbonCopy: '' })
    expect(ribbon).toMatchSnapshot()
  })

  it('does not allow custom CSS', () => {
    const ribbon = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })
    expect(ribbon).not.toHaveProp('className', 'my-custom-class')
    expect(ribbon).not.toHaveProp('style')
  })
})
