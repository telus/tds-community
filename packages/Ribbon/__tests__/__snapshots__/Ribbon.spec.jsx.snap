// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon renders 1`] = `
.c0 {
  background-color: #4b286d;
  padding: 0px 1.5rem 0px 1rem;
  position: relative;
  display: inline-block;
  max-width: 100%;
  min-width: 7.875rem;
  height: 1.5rem;
  font-weight: 500;
}

.c0:after {
  content: "";
  position: absolute;
  right: -0.3125rem;
  bottom: -0.5rem;
  width: 0;
  height: 0;
  border-right: 0.8125rem solid white;
  border-top: 1.25rem solid transparent;
  border-bottom: 1.25rem solid transparent;
}

.c0 > span {
  line-height: 1 !important;
}

<Ribbon>
  <Ribbon__StyledRibbon>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Ribbon__StyledRibbon-sc-1qcqwhs-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "background-color: #4b286d; padding: 0px 1.5rem 0px 1rem; position: relative; display: inline-block; max-width: 100%; min-width: 7.875rem; height: 1.5rem; font-weight: 500; &:after {
  content: \\"\\"; position: absolute; right: -0.3125rem; bottom: -0.5rem; width: 0; height: 0; border-right: 0.8125rem solid white; border-top: 1.25rem solid transparent; border-bottom: 1.25rem solid transparent;
} > span {
  line-height: 1 !important;
}",
            ],
          },
          "displayName": "Ribbon__StyledRibbon",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Ribbon__StyledRibbon-sc-1qcqwhs-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <Text
          block={false}
          bold={true}
          invert={true}
          size="small"
        >
          <span
            className="TDS_Typography-modules__default___KlcpP TDS_Typography-modules__small___mt5lS TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__boldFont___2YS3O TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__invertedColor___2tTa_"
          >
            2 year plan
          </span>
        </Text>
      </div>
    </StyledComponent>
  </Ribbon__StyledRibbon>
</Ribbon>
`;
